lib/kcerts/cache.go

```diff
@@ -0,0 +1,67 @@
+package kcerts
+
+import (
+	"errors"
+	"fmt"
+	"github.com/enfabrica/enkit/lib/cache"
+	"github.com/enfabrica/enkit/lib/config/marshal"
+	"path/filepath"
+)
+
+const (
+	SSHCacheKey  = "enkitssh"
+	SSHCacheFile = "ssh.json"
+)
+
+var (
+	SSHAgentNoCache      = errors.New("the cache had not existed before")
+	SSHAgentCacheInvalid = errors.New("the cache is corrupted or invalid json")
+)
+
+func FetchSSHAgentFromCache(store cache.Store) (*SSHAgent, error) {
+	sshEnkitCache, err := store.Exists(SSHCacheKey)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching cache: %w", err)
+	}
+	if sshEnkitCache == "" {
+		return nil, SSHAgentNoCache
+	}
+	agent := &SSHAgent{
+		Close: func() {},
+	}
+	if err := marshal.UnmarshalFile(filepath.Join(sshEnkitCache, SSHCacheFile), &agent); err != nil {
+		return nil, fmt.Errorf("%s: %w", SSHAgentCacheInvalid, err)
```

Comment:
ping on the comment I had added here. The SSHAgentCacheInvalid here... is pretty much just a string? I would recommend removing SSHAgentCacheInvalid. 

Basically: the function returns "an agent, and an error", an error is returned if the cache is invalid (eg, line 24 is a form of invalid cache), unless it's SSHAgentNoCache.